### PR TITLE
auto-updating bandwidth statistics

### PR DESCRIPTION
**Pull request by ultraschuppi**
requested by https://github.com/sharu-io/desktop/issues/10

### CODE REVIEW

File: src/app/components/statistics/statistics.component.html [hunk 1]
```diff
@@ -17,7 +15,7 @@ <h3>BandWidth</h3>
             <td mat-cell *matCellDef="let element"> {{element.value}} </td>
         </ng-container>
 
-        <tr mat-row *matRowDef="let row; columns: displayedColumns;"></tr>
+        <tr mat-row *matRowDef="let row; columns: displayedColumnsBandWith;"></tr>
     </table>
 <br/>
     &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
```
Comment (ColRad) [inline src/app/components/statistics/statistics.component.html]: This seems no recent change, but what are all those &nbsp; for?

File: src/app/service/statistics.service.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            this.runStatistics = (whatToShow === 'statistics');
+            if (this.runStatistics) {
+                this.updateStatsBw();
+            }
+        });
+    }
+
+    public statsBw = null;
+    private async updateStatsBw() {
+        while (this.runStatistics) {
```
Comment (ColRad) [inline src/app/service/statistics.service.ts]: Just a nitpick, but the JS way would be
`intervalId = setInteval(cb, 2000)` 
for this kind of task. An then stop via 
`clearInverval(intervalId)`

File: src/app/components/statistics/statistics.component.html (hunk 1 above)
Comment (ultraschuppi) [reply src/app/components/statistics/statistics.component.html]: you are right

File: src/app/service/statistics.service.ts (hunk 2 above)
Comment (ultraschuppi) [reply src/app/service/statistics.service.ts]: thanks, that does help! will try again

**Comment by ultraschuppi** [comment]
added some statistics about the connected peers into this PR as well...

**Comment by urbanisierung** [comment]
looks good and works as expected. merging into master

**Pull request merged**